dist/bundle.js

```diff
@@ -0,0 +1,743 @@
+  if(this.type) {
+      var canvas = d3.select('#'+this.type+'-canvas').append('svg')
+                     .attr({
+                       id: this.type+"-svg",
+                       width: this.pixelWidth,
+                       height: this.pixelHeight
+                     });
+      if(this.type === "input")
+        canvas.call(this.vectorDrag());
+
+      if(this.type === "output") {
+        // Define defs to store target image pattern
+        // Maybe figure out a better place for this code later
+        var defs = d3.select('#'+this.type+'-svg').append('defs')
+                                  .attr("id", "canvas-defs");
+        defs.append('pattern')
+            .attr({
+              "id": "tar_img",
+              "x": "0",
+              "y": "0",
+              "height": "40",
+              "width": "40"
+            });
+        d3.select('#tar_img').append('image')
+                             .attr({
+                               "x": "0",
+                               "y": "0",
+                               "width": "40",
+                               "height": "40",
+                               "xlink:href": "../public/img/target.gif"
+                             });
+      }
+      // remove this and notify eye of sauron instead
+      // updateLog(d) as example
+  }
+  else {
+    console.log("Invalid canvas type: ",this.type)
+  }
+};
+
+
+Canvas.prototype.drawProgressBar = function() {
+  var container = d3.select('#progress-container');
+      container.append('div')
+                .attr({
+                  "class": "progress-bar progress-bar-striped active",
+                  "role": "progressbar",
+                  "aria-valuenow": "0",
+                  "aria-valuemin": "0",
+                  "aria-valuemax": "100",
+                  //"style": "width: 0%",
+                  "id" : "progressbar"
+                });
+      container.append('span')
+         .attr("id", "score")
+         .text("0% Complete");
+}
+
+function updateInputVector(d){
+// redraw input vector
+  d3.select('#input-vector').remove();
+  d3.select('#input-svg').append('path')
+    .attr({
+      "stroke": "red",
+      "stroke-width":"4",
+      "d": "M 250 250 L"+d.x+" "+d.y+"z",
+      "id": 'input-vector'
+  });
+};
+
+function updateOutputVector(d) {
+  var i = applyMatrix(d.x,d.y);
+  d3.select('#output-vector').remove();
+  d3.select('#output-svg').append('path')
+    .attr({
+      "stroke": "red",
+      "stroke-width":"4",
+      //"fill": "/path/to/here",
+      "d": "M 250 250 L"+i[0]+" "+i[1]+"z",
+      "id": 'output-vector'
+  });
+};
+
+function updateTargets(d) {
+  var x = d3.selectAll("circle").attr("cx"),
+      y = d3.selectAll("circle").attr("cy"),
+      r = d3.selectAll("circle").attr("r"),
+      i = applyMatrix(d.x,d.y);
+  if (isClose(i[0],i[1],x,y,r)) {
+    d3.selectAll("circle").remove();
+    updateProgress();
+    generateTarget([[1,3],[2,0]]);
+  }
+}
+
+function updateProgress() {
```

Comment:
We need to put these functions inside of classes. although it works for right now.